.circleci/config.yml

```diff
@@ -288,25 +288,38 @@ jobs:
               circleci-agent step halt
               exit 0
             }
+
+            # Handle skip tests on labels
             if [[ $(./tools/ci/has_label.sh ci/skip-test) == "true" ]]; then
               skip "Not running tests as the PR has label: ci/skip-test"
             fi
             if [[ $(./tools/ci/has_label.sh ci/skip-e2e-test) == "true" ]]; then
               skip "Not running tests as the PR has label: ci/skip-e2e-test"
             fi
+            
+            # Handle invalid test combinations
             if [[ "<< parameters.k8sVersion >>" == "kind" && "<< parameters.target >>" != "test/e2e-universal" ]]; then
               skip "kind should only be used when testing ipv6 or with test/e2e-universal"
             fi
             if [[ "<< parameters.k8sVersion >>" != kind* && "<< parameters.target >>" == "test/e2e-universal" ]]; then
               skip "universal only runs on kind"
             fi
-            if [[ ! $(./tools/ci/has_label.sh ci/run-full-matrix) == "true" ]]; then
-              if [[ "<< pipeline.git.branch >>" != "master" && "<< pipeline.git.branch >>" != "release-"* ]]; then
-                if [[ "<< parameters.cniNetworkPlugin >>" == "calico" || "<< parameters.k8sVersion >>" == "kindIpv6" || "<< parameters.k8sVersion >>" == "v1.19.16-k3s1" || "<< parameters.arch >>" == "arm64" ]]; then
+            
+            # Handle force run tests on labels
+            if [[ $(./tools/ci/has_label.sh ci/run-full-matrix) == "true" ]]; then
+              exit 0
+            fi
+            
+            # Handle legacy tests on branch
+            if [[ "<< parameters.target >>" == "test/e2e-legacy" && "<< pipeline.git.branch >>" != "master" && "<< pipeline.git.branch >>" != "release-"* ]]; then
+              if [[ ! $(./tools/ci/has_label.sh ci/run-e2e-legacy) == "true" ]];
+                skip "we do not run legacy E2E on branch by default. To run them add ci/run-e2e-legacy label"
+              fi
+              if [[ "<< parameters.cniNetworkPlugin >>" == "calico" || "<< parameters.k8sVersion >>" == "kindIpv6" || "<< parameters.k8sVersion >>" == "v1.19.16-k3s1" || "<< parameters.arch >>" == "arm64" ]]; then
```

Comment:
Should we just remove this now? (when adding the label we should probably run the other too)

Reply:
It's either
* `run-e2e-legacy` to just run one variant of E2E (the one we were running so far)
* `run-full-matrix` to run everything 

Do you think `run-e2e-legacy` is not useful and we just use `run-full-matrix` ?

Reply:
Yes that was my thinking (would people really bother making a difference?)

Reply:
Ok, I gather more opinions. I'm dropping the `run-e2e-legacy`